File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/interactive/meeting/TeleportBuddiesRequestedMessage.scala





















// Generated on 05/03/2013 10:34:52
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait TeleportBuddiesRequestedMessage extends MessageDeserializer {
  val messageId = 6302

  protected trait Message extends BaseMessage with Serializable {
    val messageId = TeleportBuddiesRequestedMessage.messageId

    val dungeonId: Short
    val inviterId: Int
    val invalidBuddiesIds: Seq[Int]

    def serialize(buf: Buffer) {
      buf.writeShort(dungeonId)
      buf.writeInt(inviterId)
      buf.writeUShort(invalidBuddiesIds.length)
      for (entry <- invalidBuddiesIds) {
        buf.writeInt(entry)
      }
    }

  }

  protected def create(dungeonId: Short, inviterId: Int, invalidBuddiesIds: Seq[Int]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val dungeonId = buf.readShort()
    val inviterId = buf.readInt()
    val invalidBuddiesIds = {
      val builder = collection.immutable.Seq.newBuilder[Int]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readInt()
      }
      builder.result()
    }

    create(dungeonId, inviterId, invalidBuddiesIds)
  }

}

object TeleportBuddiesRequestedMessage extends TeleportBuddiesRequestedMessage {
  case class MessageImpl(dungeonId: Short, inviterId: Int, invalidBuddiesIds: Seq[Int]) extends Message
  type Target = MessageImpl

  override def create(dungeonId: Short, inviterId: Int, invalidBuddiesIds: Seq[Int]) = MessageImpl(dungeonId, inviterId, invalidBuddiesIds)

  def apply(dungeonId: Short, inviterId: Int, invalidBuddiesIds: Seq[Int]) = create(dungeonId, inviterId, invalidBuddiesIds)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/TeleportBuddiesRequestedMessage.java


// Generated on 05/08/2013 19:37:52
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class TeleportBuddiesRequestedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6302;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short dungeonId;
    public int inviterId;
    public int[] invalidBuddiesIds;
    
    public TeleportBuddiesRequestedMessage() { }
    
    public TeleportBuddiesRequestedMessage(short dungeonId, int inviterId, int[] invalidBuddiesIds) {
        this.dungeonId = dungeonId;
        this.inviterId = inviterId;
        this.invalidBuddiesIds = invalidBuddiesIds;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(dungeonId);
        buf.writeInt(inviterId);
        buf.writeUShort(invalidBuddiesIds.length);
        for (int entry : invalidBuddiesIds) {
            buf.writeInt(entry);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        dungeonId = buf.readShort();
        if (dungeonId < 0)
            throw new RuntimeException("Forbidden value on dungeonId = " + dungeonId + ", it doesn't respect the following condition : dungeonId < 0");
        inviterId = buf.readInt();
        if (inviterId < 0)
            throw new RuntimeException("Forbidden value on inviterId = " + inviterId + ", it doesn't respect the following condition : inviterId < 0");
        int limit = buf.readUShort();
        invalidBuddiesIds = new int[limit];
        for (int i = 0; i < limit; i++) {
            invalidBuddiesIds[i] = buf.readInt();
        }
    }
    
}
